js/viz/chart.js

```diff
@@ -371,31 +371,37 @@ function collectMarkersInfoBySeries(allSeries, filteredSeries, argAxis) {
     return { series, overloadedSeries };
 }
 
-function applyAutoHidePointMarkers(allSeries, filteredSeries, overloadedSeries, argAxis) {
-    const argAxisType = argAxis.getOptions().type;
-    filteredSeries.forEach(s => {
-        const seriesIndex = allSeries.indexOf(s);
-        s.autoHidePointMarkers = false;
-        const tickCount = argAxis.getTicksValues().majorTicksValues.length;
-        if(s.autoHidePointMarkersEnabled() && (argAxisType === DISCRETE || overloadedSeries[seriesIndex].pointsCount > tickCount)) {
-            for(const index in overloadedSeries[seriesIndex]) {
-                const i = parseInt(index);
-                if(isNaN(i) || overloadedSeries[seriesIndex].total / overloadedSeries[seriesIndex].continuousSeries < 3) {
-                    continue;
-                }
-                if(i === seriesIndex) {
-                    if(overloadedSeries[i][i] * 2 >= overloadedSeries[i].pointsCount) {
-                        s.autoHidePointMarkers = true;
-                        break;
-                    }
-                } else if(overloadedSeries[seriesIndex].total >= overloadedSeries[seriesIndex].pointsCount) {
-                    s.autoHidePointMarkers = true;
-                    break;
-                }
+const isOverlay = (currentPoint, overlayPoint, radiusPoint) => {
+    const isPointHitLeftBorder = overlayPoint.x - radiusPoint <= currentPoint.x;
+    const isPointHitRightBorder = overlayPoint.x + radiusPoint >= currentPoint.x;
+    const isPointHitTopBorder = overlayPoint.y - radiusPoint <= currentPoint.y;
+    const isPointHitBottomBorder = overlayPoint.y + radiusPoint >= currentPoint.y;
+    const isPointOverlayHorizontally = isPointHitLeftBorder && isPointHitRightBorder;
+    const isPointOverlayVertically = isPointHitTopBorder && isPointHitBottomBorder;
+
+    return isPointOverlayHorizontally && isPointOverlayVertically;
+};
+
+const isPointOverlapped = (currentPoint, overlappingPoints, skipComparingSamePoints) => {
```

Comment:
1. We are not sure that these points are overlapping. Moreover, `points` would be just more compact
2. `skipComparingSamePoints` -> `skipSamePointsComparing`
```suggestion
const isPointOverlapped = (currentPoint, points, skipSamePointsComparing) => {
```

Reply:
Fixed